orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -6599,3 +6602,146 @@ def run_step(model, attn_weight):
     assert to_attr.name == "to"
     to_value = to_attr.i
     assert to_value == pytorch_type_to_onnx_dtype(softmax_compute_type), "Cast to attribute is not as expected"
+
+
+@pytest.mark.parametrize("memory_optimization_level", [None, 0, 1, 2])
+def test_auto_enable_layerwise_recompute_local_import(memory_optimization_level, caplog):
+    from torch.utils.checkpoint import checkpoint
+
+    original_val = os.environ.get("ORTMODULE_MEMORY_OPT_LEVEL", None)
+
+    if memory_optimization_level is not None:
+        os.environ["ORTMODULE_MEMORY_OPT_LEVEL"] = str(memory_optimization_level)
+    else:
+        if original_val is not None:
+            del os.environ["ORTMODULE_MEMORY_OPT_LEVEL"]
+
+    class SampleModel(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.layer1 = nn.Linear(10, 10)
+            self.layer2 = nn.Linear(10, 10)
+
+        def forward(self, x):
+            # Checkpointing the first layer
+            x = checkpoint(self.layer1, x)
+            x = nn.ReLU()(x)
+            # The second layer is not checkpointed
+            x = self.layer2(x)
+            return x
+
+    model = SampleModel().cuda()
+    input = torch.randn(1, 10).cuda()
+    model = ORTModule(model, DebugOptions(log_level=LogLevel.INFO))
+
+    expected_level_before_run = 0 if memory_optimization_level is None else memory_optimization_level
+
+    assert (
+        model._torch_module._execution_manager._training_manager._runtime_options.memory_optimization_level
+        == expected_level_before_run
+    )
+
+    # Forward pass
+    is_export_failed = False
+    try:
+        _ = model(input)
+    except Exception as e:
```

Comment:
Consider re-throwing underlying error.

Reply:
This should be caught otherwise the test will crash on some versions of PyTorch . Use a context manager to catch potential failures.